google/cloud/storage/internal/service_account_credentials.h

```diff
@@ -0,0 +1,176 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_SERVICE_ACCOUNT_CREDENTIALS_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_SERVICE_ACCOUNT_CREDENTIALS_H_
+
+#include "google/cloud/storage/credentials.h"
+#include "google/cloud/storage/internal/credential_constants.h"
+#include "google/cloud/storage/internal/curl_request.h"
+#include "google/cloud/storage/internal/nljson.h"
+#include "google/cloud/storage/internal/openssl_util.h"
+#include <chrono>
+#include <condition_variable>
+#include <ctime>
+#include <iostream>
+#include <mutex>
+#include <sstream>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+/**
+ * A C++ wrapper for Google's Service Account Credentials.
+ *
+ * Takes a JSON object representing the contents of a service account keyfile,
+ * and uses Google's OAuth2 service to obtain an access token.
+ *
+ * @warning
+ * The current implementation is a placeholder to unblock development of the
+ * Google Cloud Storage client libraries. There is substantial work needed
+ * before this class is complete, in fact, we do not even have a complete set of
+ * requirements for it.
+ *
+ * @see
+ *   https://developers.google.com/identity/protocols/OAuth2ServiceAccount
+ *   https://tools.ietf.org/html/rfc7523
+ *
+ * @tparam HttpRequestType a dependency injection point to make HTTP requests.
```

Comment:
You are missing the `@tparam` documentation for `ClockType`, that will break the build because Doxygen warnings are errors.

Reply:
Ah, shoot. I added this typename in after-the-fact for testing and forgot to add the docstring for it. Thanks, fixed.